src/python/WMCore/Credential/Proxy.py

```diff
@@ -266,6 +266,39 @@ def getSubjectFromCert(self, certFile = None):
 
         return subject
 
+    def getDatesFromCert(self, certFile = None):
+        """
+        Get the dates from cert file.
+        """
+        dates = {}
+
+        if not certFile:
+            certFile = self.getProxyFilename()
+
+        datesFromCertCmd = 'openssl x509 -in '+certFile+' -dates -noout'
+        datesResult, _, retcode = execute_command(self.setEnv(datesFromCertCmd), self.logger, self.commandTimeout)
+
+        if retcode == 0:
+            dates["notBefore"] = re.search("(?<=notBefore=).*", datesResult).group(0)
+            dates["notAfter"] = re.search("(?<=notAfter=).*", datesResult).group(0)
+
+            possibleFormats = ['%b  %d  %H:%M:%S %Y %Z', '%b %d %H:%M:%S %Y %Z']
+            exptime = None
+            for frmt in possibleFormats:
+                try:
+                    exptime = datetime.strptime(dates["notAfter"], frmt)
+                except ValueError:
+                    pass #try next format
+                if not exptime:
```

Comment:
Better add this `if` block outside the `for` loop, meaning none of the time formats were accepted.
Thus you can keep the `try/except/else` block.